skrub/_table_vectorizer.py

```diff
@@ -294,18 +252,24 @@ class TableVectorizer(TransformerMixin, BaseEstimator):
         Mixing the two is not supported.
 
     auto_cast : bool, default=True
-        If set to `True`, will try to convert each column to the best possible
-        data type (dtype).
-
-    impute_missing : {'auto', 'force', 'skip'}, default='auto'
+        If set to ``True``, calling fit or transform will try to convert each column
+        to the "optimal" dtype for scikit-learn estimators.
+        The main heuristics are the following:
+        - pandas extension dtypes conversion to numpy dtype
+        - datetime conversion using ``skrub.to_datetime``
+        - numeric conversion using ``pandas.to_numeric``
+        - numeric columns with missing values are converted to float to input np.nan
+        - categorical columns with missing values are imputed with 'missing' if
+          ``impute_missing_categories`` is ``True``.
+        - categorical columns dtypes are updated with the new entries (if any)
+          during transform.
+
+    impute_missing_categories : bool, default=True
```

Comment:
I'm not sure I understand the behavior around missing values. for categories we replace missing values with "missing", and for strings (which get handled as categories when low cardinality) we do the opposite, replacing "false missing" values such as the empty string with np.nan

also missing values in columns with the `string` dtype raise an error
```
>>> df = pd.DataFrame({"astring": ["Aa", "Bb", None]}).convert_dtypes()
>>> TableVectorizer().fit_transform(df) #TypeError
```

Reply:
Yes, thank you for asking. It doesn't make much sense to me either. I simplified the logic without altering it as much as possible. We have to discuss with the rest of the time some of the choices made for impute such as this one.

Reply:
that makes sense, we can open another issue

Reply:
Here it is: https://github.com/skrub-data/skrub/issues/837